mako/cmd.py

```diff
@@ -51,13 +51,26 @@ def cmdline(argv=None):
         default=None,
         help="Write to file upon successful render instead of stdout",
     )
-    parser.add_argument("input", nargs="?", default="-")
+    parser.add_argument(
+        "--strip-shebang", "-s", action="store_true",
+        help="Strip a shebang in the first line of given input file",
+    )
+    parser.add_argument("input", nargs="?", default="-",
+                        help="file to parse as template. default is stdin.")
+    parser.add_argument("args", nargs="*",
+                        help=("arguments passed to template's script. "
+                              "use '-- args...' when first arg "
+                              "starts with - or "))
 
     options = parser.parse_args(argv)
 
     output_encoding = options.output_encoding
     output_file = options.output_file
 
+    # strip away mako's original args, so the template can get new args
+    saved_sys_argv = sys.argv
+    sys.argv = [options.input] + options.args
```

Comment:
I would not favor modifying sys.argv or having templates refer to sys.    right now --var will pass a single variable to the template, I would favor an option that can be passed an expansion if this use case is desired.

Reply:
i've updated the patch to only updates further arguments if present.
this is also for the use-case of importing `argparse` within the template

Reply:
ok but argparse doesnt read sys.argv when it's imported, it reads it by default when you call upon parser.parse_args() where you can pass the arguments explicitly, there's no requirement to use sys.argv.  

I get that having sys.argv look normal is convenient but it seems better to have sys.argv remain the "real" sys.argv and then some other variable can be the template's argv.   We can send it to the template by default even, like "template_argv".

Reply:
ok, i can add an option to enable this arg-replacement then. if this option is not active, we  can pass the args as `template_args` variable so one can invoke `parse_args` with that. an alternative to these options could be installing another entrypoint, maybe called `mako-exec` alongside the `mako-render` :)